src/clr-angular/forms/datepicker/utils/date-utils.ts

```diff
@@ -41,3 +41,15 @@ export function parseToFourDigitYear(year: number): number {
   }
   return result;
 }
+
+export function datesAreEqual(date1: Date, date2: Date) {
+  if (date1 && date2) {
```

Comment:
I lean towards it should throw to prevent the function from masking a mistake, ex I pass in a string/number accidentally and still got a value back as if nothing was wrong. What does the team/codebase tend to lean towards? Throw with invalid types or defend/check against invalid types on params? Actually makes me kinda question why I had the if block to begin with... I'll double check that.

Reply:
I'd like it to just return `false` in this case at least, since this is really a runtime check. If the developer explicitly calls it incorrectly TS will warn them, but we can't always be 100% certain of the values passed into this function.